lib/CustomFields/components/CustomFieldsForm/CustomFieldsForm.js

```diff
@@ -0,0 +1,377 @@
+      <Col xs>
+        <ExpandAllButton
+          accordionStatus={accordions}
+          onToggle={handleExpandAll}
+        />
+      </Col>
+    </Row>
+  );
+
+  const renderPaneFooter = () => {
+    const buttonsDisabled = submitting || (pristine && !fieldsToDelete.length);
+
+    return (
+      <PaneFooter
+        renderStart={
+          <Button
+            marginBottom0
+            onClick={onResetClick}
+            disabled={buttonsDisabled}
+          >
+            <FormattedMessage id="stripes-smart-components.customFields.cancel" />
+          </Button>
+        }
+        renderEnd={
+          <Button
+            marginBottom0
+            onClick={onSaveClick}
+            disabled={buttonsDisabled}
+          >
+            <FormattedMessage id="stripes-smart-components.customFields.save" />
+          </Button>
+        }
+      />
+    );
+  };
+
+  const onToggleAccordion = ({ id }) => {
+    setAccordions(currentAccordions => ({
+      ...currentAccordions,
+      [id]: !currentAccordions[id],
+    }));
+  };
+
+  const renderFieldAccordions = () => (
+    <div className={styles.accordionsWrapper}>
+      <AccordionSet accordionStatus={accordions} onToggle={onToggleAccordion}>
+        {customFields.map(cf => {
+          const {
+            id,
+            values,
+          } = cf;
+
+          return (
+            <FieldAccordion
+              id={id}
+              key={id}
+              deleteCustomField={onDeleteClick(id)}
+              editMode
+              onChange={onFieldUpdate(cf.id)}
+              fieldData={{ ...values }}
+            />
+          );
+        })}
+      </AccordionSet>
+    </div>
+  );
+
+  const renderAccordionName = () => {
+    return (
+      <FormattedMessage id="stripes-smart-components.customFields.recordAccordion.defaultName">
+        {defaultName => (
+          <TextField
+            value={defaultName}
+            inputClass={styles.accordionNameInput}
+            className={styles.inputWrapper}
+            disabled
+            label={
+              <span className={styles.accordionNameLabel}>
+                <FormattedMessage id="stripes-smart-components.customFields.settings.sectionTitle" />
+              </span>
+            }
+          />
+        )}
+      </FormattedMessage>
+    );
+  };
+
+  const getFieldTypesOptions = () => {
+    return Object.values(fieldTypes).map(fieldType => ({
+      label: fieldTypesLabels[fieldType],
+      value: fieldType,
+    }));
+  };
+
+  const formattedFieldsToDeleteData = useMemo(() => fieldsToDelete.reduce((acc, cf) => ([
+    ...acc,
+    {
+      index: cf.index,
+      id: cf.data.id,
+      name: cf.data.values.name,
+    }
+  ]), []), [fieldsToDelete]);
+
+  return (
+    <Fragment>
+      <Paneset>
+        <Pane
+          paneTitle={<FormattedMessage id="stripes-smart-components.customFields.editCustomFields" />}
+          firstMenu={renderFirstMenu()}
```

Comment:
Consider adding `<PaneHeader>` component to the `renderHeader` prop and move these props to the header.